### PR TITLE
Refactor publishing

### PR DESCRIPTION
**Pull request by Janaka-Steph**
workflow_dispatch => build/check + send to artifacts
tag => build/check + publish in GH release
And release scripts

### CODE REVIEW

File: .github/workflows/workflow-dispatch.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+on:
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+      tags:
+        required: false
+        description: 'Test scenario tags'
```
Comment (tiero) [inline .github/workflows/workflow-dispatch.yml:10]:
```suggestion
```

not needed

**Comment by tiero** [comment]
I do think we should do the following:

- for internal releases (Testflight or APK) we still rely on local fastlane for ios and run workflow dispatch for APK that posts to artifacts  

- for promoting a release we use a script to increment the version & build of Gradle and CF Build string and commit to master
- then we git tag from there and push 
- release.yml runs, builds APK and create & upload to Github Release and push package json cooridnate with tag

- we manually sign APK and push to Play Store
- we manually promote on AppStore Connect the already uplaoded build

File: scripts/tag.sh [hunk 2]
```diff
@@ -5,8 +5,6 @@ set -e
...
 read -r TAG
 
-git add .
-git commit -m "$TAG"
-git tag "$TAG"
-git push origin "$TAG"
-git push origin master
\ No newline at end of file
+echo "Bump iOS and Android"
+fastlane bump
```
Comment (Janaka-Steph) [inline scripts/tag.sh:10]: The TAG is useless here. What about moving `fastlane bump` in a npm script?

Comment (tiero) [reply scripts/tag.sh:10]: the pkg.json version is used by fastlane bump, so must be incremented via yarn version, which also does git tag under the hood

**Pull request merged**